p2p_wallet/Scenes/Main/Swap/State/JupiterSwapStateMachine.swift

```diff
@@ -1,21 +1,68 @@
-import Send // TODO: I will extract StateMachine in core module inside Key App Kit
 import Combine
 
-actor JupiterSwapStateMachine: StateMachine {
+actor JupiterSwapStateMachine {
+    // MARK: - Nested type
+    
+    /// The cache that handle currentTask and currentAction
+    /// Must be actor to make sure that currentTask and currentAction are thread-safe
+    private actor Cache {
+        /// Current executing task
+        fileprivate var currentTask: Task<JupiterSwapState, Never>?
+        /// Save the current task
+        fileprivate func saveCurrentTask(_ task: Task<JupiterSwapState, Never>?) {
+            currentTask = task
+        }
+    }
+    
+    // MARK: - Properties
+
     private nonisolated let stateSubject: CurrentValueSubject<JupiterSwapState, Never>
+    private nonisolated let cache = Cache()
+
+    // MARK: - Public properties
 
     nonisolated var statePublisher: AnyPublisher<JupiterSwapState, Never> { stateSubject.eraseToAnyPublisher() }
     nonisolated var currentState: JupiterSwapState { stateSubject.value }
 
     nonisolated let services: JupiterSwapServices
 
+    // MARK: - Initializer
+
     init(initialState: JupiterSwapState, services: JupiterSwapServices) {
         stateSubject = .init(initialState)
         self.services = services
     }
+    
+    // MARK: - Accept function
 
     @discardableResult
-    func accept(action: JupiterSwapAction) async -> JupiterSwapState {
+    nonisolated func accept(
+        action newAction: JupiterSwapAction,
+        waitForPreviousActionToComplete: Bool
+    ) async -> JupiterSwapState {
```

Comment:
change accept function to nonisolated, so it can receive multiple signals from outside. For example: 
- Accept `firstAction`.
- In the mean time accept `secondAction` in parallel.
- It will cancel or wait the `firstAction`, depend on flag `waitForPreviousActionToComplete`.